plugins/salesforce/komand_salesforce/tasks/monitor_users/task.py

```diff
@@ -0,0 +1,153 @@
+import insightconnect_plugin_runtime
+from .schema import MonitorUsersInput, MonitorUsersOutput, MonitorUsersState, Component
+
+# Custom imports below
+from datetime import datetime, timedelta, timezone
+from komand_salesforce.util.exceptions import ApiException
+from komand_salesforce.util.helpers import clean, convert_to_camel_case
+
+
+class MonitorUsers(insightconnect_plugin_runtime.Task):
+    USER_LOGIN_QUERY = "SELECT LoginTime, UserId, LoginType, LoginUrl, SourceIp, Status, Application, Browser FROM LoginHistory WHERE LoginTime >= {start_timestamp} AND LoginTime < {end_timestamp}"
+    USERS_QUERY = "SELECT Id, FirstName, LastName, Email, Alias, IsActive FROM User WHERE UserType = 'Standard'"
+    UPDATED_USER_QUERY = "SELECT Id, FirstName, LastName, Email, Alias, IsActive FROM User WHERE Id = '{user_id}' AND UserType = 'Standard'"
+    USERS_NEXT_PAGE_ID = "users_next_page_id"
+    USER_LOGIN_NEXT_PAGE_ID = "user_login_next_page_id"
+    LAST_USER_UPDATE_COLLECTION_TIMESTAMP = "last_user_update_collection_timestamp"
+    NEXT_USER_COLLECTION_TIMESTAMP = "next_user_collection_timestamp"
+    NEXT_USER_LOGIN_COLLECTION_TIMESTAMP = "next_user_login_collection_timestamp"
+    LAST_USER_LOGIN_COLLECTION_TIMESTAMP = "last_user_login_collection_timestamp"
+
+    def __init__(self):
+        super(self.__class__, self).__init__(
+            name="monitor_users",
+            description=Component.DESCRIPTION,
+            input=MonitorUsersInput(),
+            output=MonitorUsersOutput(),
+            state=MonitorUsersState(),
+        )
+
+    # pylint: disable=unused-argument
+    def run(self, params={}, state={}):  # noqa: C901
```

Comment:
I think this run function requires an additional try/except to ensure all possible exceptions are captured and bubbled out as a return from the function, otherwise any exception that is not captured e.g. by the apiexception will be lost as there is no way for these to be passed back?

Reply:
@dlaverty-r7 I added an additional try/except.